lib/ansible/modules/windows/win_domain_user.py

```diff
@@ -0,0 +1,361 @@
+    required: false
+    choices: [ 'always', 'on_create' ]
+    default: always
+
+  password_expired:
+    description:
+      - C(yes) will require the user to change their password at next login.
+        C(no) will clear the expired password flag. This is mutually exclusive
+        with I(password_never_expires).
+    required: false
+    choices: [ 'yes', 'no' ]
+    default: null
+
+  password_never_expires:
+    description:
+      - C(yes) will set the password to never expire.  C(no) will allow the
+        password to expire. This is mutually exclusive with I(password_expired)
+    required: false
+    choices: [ 'yes', 'no' ]
+    default: null
+
+  user_cannot_change_password:
+    description:
+      - C(yes) will prevent the user from changing their password.  C(no) will
+        allow the user to change their password.
+    required: false
+    choices: [ 'yes', 'no' ]
+    default: null
+
+  firstname:
+    description:
+      - Configures the user's first name (given name)
+    required: false
+    default: null
+
+  surname:
+    description:
+      - Configures the user's last name (surname)
+    required: false
+    default: null
+
+  company:
+    description:
+      - Configures the user's company name
+    required: false
+    default: null
+
+  upn:
+    description:
+      - Configures the User Principal Name (UPN) for the account. This is not
+        required, but is best practice to configure for modern versions of
+        Active Directory. The format is "<username>@<domain>".
+    required: false
+    default: null
+
+  email:
+    description:
+      - Configures the user's email address. This is a record in AD and does
+        not do anything to configure any email servers or systems.
+    required: false
+    default: null
+
+  street:
+    description:
+      - Configures the user's street address
+    required: false
+    default: null
+
+  city:
+    description:
+      - Configures the user's city
+    required: false
+    default: null
+
+  state_province:
+    description:
+      - Configures the user's state or province
+    required: false
+    default: null
+
+  postal_code:
+    description:
+      - Configures the user's postal code / zip code
+    required: false
+    default: null
+
+  country:
+    description:
+      - Configures the user's country code. Note that this is a two-character
+        ISO 3166 code.
+    required: false
+    default: null
+
+  path:
+    description:
+      - Container or OU for the new user; if you do not specify this, the
+        user will be placed in the default container for users in the domain.
+        Setting the path is only available when a new user is created;
+        if you specify a path on an existing user, the user's path will not
+        be updated - you must delete (e.g., state=absent) the user and
+        then re-add the user with the appropriate path.
+    required: false
+    default: null
+
+author:
+    - "Nick Chandler (@nwchandler)"
```

Comment:
We tend to not add quotes where they are no needed in YAML.
Here they are not needed.